src/core/stdcpp/vector.d

```diff
@@ -0,0 +1,248 @@
+/**
+ * D header file for interaction with C++ std::vector.
+ *
+ * Copyright: Copyright (c) 2018 D Language Foundation
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Guillaume Chatelet
+ *            Manu Evans
+ * Source:    $(DRUNTIMESRC core/stdcpp/vector.d)
+ */
+
+module core.stdcpp.vector;
+
+///////////////////////////////////////////////////////////////////////////////
+// std::vector declaration.
+//
+// Current caveats :
+// - mangling issues still exist
+// - won't work with custom allocators
+// - missing noexcept
+// - iterators are implemented as pointers
+// - no reverse_iterator nor rbegin/rend
+// - nothrow @trusted @nogc for most functions depend on knowledge
+//   of T's construction/destruction/assignment semantics
+///////////////////////////////////////////////////////////////////////////////
+
+import core.stdcpp.allocator;
+
+enum DefaultConstruct { value }
+
+/// Constructor argument for default construction
+enum Default = DefaultConstruct();
+
+extern(C++, "std"):
+
+extern(C++, class) struct vector(T, Alloc = allocator!T)
+{
+    static assert(!is(T == bool), "vector!bool not supported!");
+
+    alias value_type = T;
+    alias allocator_type = Alloc;
+    alias pointer = T*;
+    alias const_pointer = const(T)*;
+    alias difference_type = ptrdiff_t;
+    alias size_type = size_t;
+
+//    alias iterator = pointer;
+//    alias const_iterator = const_pointer;
+//    alias reverse_iterator
+//    alias const_reverse_iterator
+
+    alias as_array this;
```

Comment:
I see why you'd go this way - I did the same in my local port. But I'm not sure it's suitable for druntime.

Reply:
They're all like this. It makes these types interact with D in any meaningful way.
What would you suggest?

Reply:
To let users call `as_array()` themselves at first, see if the convenience is worth it. It will be easy to add, it's painful to remove.